lib/cosmos/system/system.rb

```diff
@@ -507,12 +509,16 @@ def load_configuration(name = nil)
             update_config(@initial_config)
           else
             # Look for the requested configuration in the saved configurations
-            configuration_directory = find_configuration(name)
-            if configuration_directory
+            configuration = find_configuration(name)
+            if configuration
               # We found the configuration requested. Reprocess the system.txt
               # and reload the packets
               begin
-                process_file(File.join(configuration_directory, 'system.txt'), configuration_directory)
+                unless File.directory?(configuration)
+                  # Zip file configuration so unzip and reset configuration path
+                  configuration = unzip(configuration)
```

Comment:
Need to prevent repeated unziping of the same configuration.   Probably just check if TMP folder already exists.